spanner/tests/system.py

```diff
@@ -492,24 +492,29 @@ def _row_data(max_index):
             yield [
                 index,
                 'First%09d' % (index,),
-                'Last09%d' % (index),
+                'Last%09d' % (max_index - index),
                 'test-%09d@example.com' % (index,),
             ]
 
-    def _set_up_table(self, row_count):
-        retry = RetryInstanceState(_has_all_ddl)
-        retry(self._db.reload)()
+    def _set_up_table(self, row_count, db=None):
 
-        session = self._db.session()
+        if db is None:
+            db = self._db
+            retry = RetryInstanceState(_has_all_ddl)
+            retry(db.reload)()
+
+        session = db.session()
         session.create()
         self.to_delete.append(session)
 
-        with session.transaction() as transaction:
-            transaction.delete(self.TABLE, self.ALL)
+        def _unit_of_work(transaction, test):
+            transaction.delete(test.TABLE, test.ALL)
             transaction.insert(
-                self.TABLE, self.COLUMNS, self._row_data(row_count))
+                test.TABLE, test.COLUMNS, test._row_data(row_count))
+
+        committed = session.run_in_transaction(_unit_of_work, test=self)
```

Comment:
Because `run_in_transaction` gets us [retry-on-abort](https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.v1#google.spanner.v1.Spanner.Commit) semantics for `Commit`.